libminifi/src/core/yaml/YamlConnectionParser.cpp

```diff
@@ -0,0 +1,181 @@
+      return destUUID.value();
+    }
+    logger_->log_error("Invalid destination id value: %s.", destination_id_node.as<std::string>());
+    throw std::invalid_argument("Invalid destination id");
+  }
+  // we use the same logic as above for resolving the source processor
+  // for looking up the destination processor in absence of a processor id
+  checkRequiredField(&connectionNode_, "destination name", logger_, CONFIG_YAML_CONNECTIONS_KEY);
+  std::string connectionDestProcName = connectionNode_["destination name"].as<std::string>();
+  const utils::optional<utils::Identifier> destUUID = utils::Identifier::parse(connectionDestProcName);
+  if (destUUID && parent_->findProcessorById(destUUID.value())) {
+    // the destination name is a remote port id, so use that as the dest id
+    logger_->log_debug("Using 'destination name' containing a remote port id to match the destination for connection '%s': destination name => [%s]", name_, connectionDestProcName);
+    return destUUID.value();
+  }
+  // look the processor up by name
+  auto destProcessor = parent_->findProcessorByName(connectionDestProcName);
+  if (NULL != destProcessor) {
+    logger_->log_debug("Using 'destination name' to match destination with same name for connection '%s': destination name => [%s]", name_, connectionDestProcName);
+    return destProcessor->getUUID();
+  }
+  // we ran out of ways to discover the destination processor
+  const std::string error_msg = "Could not locate a destination with name " + connectionDestProcName + " to create a connection";
+  logger_->log_error(error_msg.c_str());
+  throw std::invalid_argument(error_msg);
+}
+
+uint64_t YamlConnectionParser::getFlowFileExpirationFromYaml() const {
+  const YAML::Node expiration_node = connectionNode_["flowfile expiration"];
+  if (!expiration_node) {
+    logger_->log_debug("parseConnection: flowfile expiration is not set, assuming 0 (never expire)");
+    return 0;
+  }
+  uint64_t expirationDuration = 0;
+  TimeUnit unit;
+  const std::string flowfile_expiration_str = expiration_node.as<std::string>();
+  if (!core::Property::StringToTime(flowfile_expiration_str, expirationDuration, unit) || !core::Property::ConvertTimeUnitToMS(expirationDuration, unit, expirationDuration)) {
+    // We should throw here, but we do not.
+    // The reason is that our parser only accepts time formats that consists of a number and
+    // a unit, but users might use this field populated with a "0" (and no units).
+    // We cannot correct this, because there is no API contract for the config, we need to support
+    // all already-supported configuration files.
+    // This has the side-effect of allowing values like "20 minuites" and silently defaulting to 0.
+    logger_->log_debug("parseConnection: flowfile expiration => [%d]", expirationDuration);
```

Comment:
> Added a log to the not found branch and an explanation that 0 means flowfiles never expire.

We still don't log `expirationDuration` when the time parsing is successful, only when it fails.

Reply:
Ah, I though you wanted me to log that the node was missing. Added logging now every time this function is run.